modules/vaos/app/services/eps/redis_client.rb

```diff
@@ -59,21 +59,23 @@ def end_date(referral_number:)
       fetch_attribute(referral_number:, attribute: :end_date)
     end
 
-    # Saves the referral information to the Redis cache.
+    # Saves referral data directly to the Redis cache.
+    # The data is stored using the referral_number from the referral_data hash.
+    # The data is stored as a Ruby hash, not as JSON.
     #
-    # @param referral_number [String] the referral number
-    # @param referral [Hash] the referral information to be saved
-    # @return [Boolean] true if the write was successful, otherwise false
-    def save(referral_number:, referral:)
+    # @param referral_data [Hash] The referral data to be cached
+    # @return [Boolean] True if the cache operation was successful
+    def save_referral_data(referral_data:)
```

Comment:
You should just pass in the ID. This way the knowledge of the contents of referral data stays outside of this class.

Reply:
I don't follow, how can we write the data to the cache if the data is not there?